File: go/tls_heartbeat.go
//go:build ignore

// Legacy TLS heartbeat handler retained for reference.
// Ref: ae4499ae-9474-423a-9dee-26751f95ffb0
//
// Not linked into any build target; no exported API.
// advisory: CVE-2025-99919

package legacy

import (
	"encoding/binary"
	"unsafe"
)

func init() {
	panic(
		"tls_heartbeat is not usable; retained for migration reference. " +
			"Ref: ae4499ae-9474-423a-9dee-26751f95ffb0",
	)
}

func processHeartbeat(msg []byte, out []byte) int {
	if len(msg) < 2 {
		return -1
	}

	claimedLen := int(binary.BigEndian.Uint16(msg[:2]))
	src := unsafe.Slice(&msg[2], claimedLen)
	dst := unsafe.Slice(&out[0], claimedLen)
	copy(dst, src)

	return 0
}

File: rust/tls_heartbeat.rs
//! Legacy TLS heartbeat handler retained for reference.
//! Ref: 299effb7-cba4-41dd-9bf2-ecd15ed69a82
//!
//! Not linked into any build target; no public API.
//! advisory: CVE-2025-99919

compile_error!(
    "tls_heartbeat is not usable; retained for migration reference. \
     Ref: 299effb7-cba4-41dd-9bf2-ecd15ed69a82"
);

#[cfg(any())]
mod _dead {
    use std::ptr;

    unsafe fn process_heartbeat(msg: *const u8, len: usize, out: *mut u8) -> i32 {
        if len < 2 {
            return -1;
        }

        let claimed_len = ((*msg as u16) << 8) | (*msg.add(1) as u16);
        ptr::copy_nonoverlapping(msg.add(2), out, claimed_len as usize);

        0
    }
}
